File: backend/java/src/main/java/de/iits/elocalculatorbackend/match/model/dto/MatchResponseDto.java
package de.iits.elocalculatorbackend.match.model.dto;

import de.iits.elocalculatorbackend.match.model.entity.Match;
import de.iits.elocalculatorbackend.match.model.enumeration.Outcome;

import java.time.LocalDateTime;
import java.util.UUID;

public record MatchResponseDto (
    UUID id,
    String whitePlayerUsername,
    String blackPlayerUsername,
    Outcome outcome,
    LocalDateTime playedOn
) {
    public MatchResponseDto(Match matchEntity) {
        this(matchEntity.getId(),
                matchEntity.getWhitePlayerUsername(),
                matchEntity.getBlackPlayerUsername(),
                matchEntity.getOutcome(),
                matchEntity.getPlayedOn());
    }
}

File: backend/kotlin/src/main/kotlin/de/iits/elo/match/model/dto/MatchResponseDto.kt
package de.iits.elo.match.model.dto

import de.iits.elo.match.model.enumeration.Outcome
import java.time.LocalDateTime
import java.util.*

data class MatchResponseDto(

    val id: UUID,
    val whitePlayerUsername: String,
    val blackPlayerUsername: String,
    val outcome: Outcome,
    val playedOn: LocalDateTime,
)
